File: cielo24_Java/cielo24/src/cielo24/Enums.java
package cielo24;

import com.google.gson.annotations.SerializedName;

public class Enums {

    public enum TaskType {
        JOB_CREATED,
        JOB_DELETED,
        JOB_ADD_MEDIA,
        JOB_ADD_TRANSCRIPT,
        JOB_PERFORM_TRANSCRIPTION,
        JOB_PERFORM_PREMIUM_SYNC,
        JOB_UPDATE_ELEMENTLIST,
        JOB_GET_TRANSCRIPT,
        JOB_GET_CAPTION,
        JOB_GET_ELEMENTLIST
    }

    public enum ErrorType {
        LOGIN_INVALID,
        ACCOUNT_EXISTS,
        ACCOUNT_DOES_NOT_EXIST,
        ACCOUNT_UNPRIVILEGED,
        BAD_API_TOKEN,
        INVALID_QUERY,
        INVALID_OPTION,
        INVALID_URL,
        MISSING_PARAMETER,
        NOT_IMPLEMENTED,
        ITEM_NOT_FOUND,
        INVALID_RETURN_HANDLERS,
        NOT_PARENT_ACCOUNT,
        NO_CHILDREN_FOUND,
        UNHANDLED_ERROR
    }

    public enum JobStatus {
        @SerializedName("Authorizing")
        AUTHORIZING,
        @SerializedName("Pending")
        PENDING,
        @SerializedName("In Process")
        IN_PROCESS,
        @SerializedName("Complete")
        COMPLETE,
        @SerializedName("Media Failure")
        MEDIA_FAILURE,
        @SerializedName("Reviewing")
        REVIEWING;

        @Override
        public String toString() {
            return Utils.getSerializedName(this);
        }
    }

    public enum TaskStatus {
        COMPLETE,
        INPROGRESS,
        ABORTED,
        FAILED
    }

    public enum Priority {
        ECONOMY,
        STANDARD,
        PRIORITY,
        CRITICAL
    }

    public enum Fidelity {
        MECHANICAL,
        PREMIUM,
        PROFESSIONAL
    }

    public enum CaptionFormat {
        SRT,
        SBV,
        SCC,
        DFXP,
        QT,
        TRANSCRIPT,
        TWX,
        TPM,
        WEB_VTT,
        ECHO
    }

    public enum TokenType {
        @SerializedName("word")
        WORD,
        @SerializedName("punctuation")
        PUNCTUATION,
        @SerializedName("sound")
        SOUND;

        @Override
        public String toString() {
            return Utils.getSerializedName(this);
        }
    }

    public enum Tag {
        UNKNOWN,
        INAUDIBLE,
        CROSSTALK,
        MUSIC,
        NOISE,
        LAUGH,
        COUGH,
        FOREIGN,
        BLANK_AUDIO,
        APPLAUSE,
        BLEEP,
        ENDS_SENTENCE
    }

    public enum SpeakerId {
        @SerializedName("no")
        NO,
        @SerializedName("number")
        NUMBER,
        @SerializedName("name")
        NAME;

        @Override
        public String toString() {
            return Utils.getSerializedName(this);
        }
    }

    public enum SpeakerGender {
        UNKNOWN,
        MALE,
        FEMALE
    }

    public enum Case {
        @SerializedName("upper")
        UPPER,
        @SerializedName("lower")
        LOWER,
        @SerializedName("")
        UNCHANGED;

        @Override
        public String toString() {
            return Utils.getSerializedName(this);
        }
    }

    public enum LineEnding {
        UNIX,
        WINDOWS,
        OSX
    }

    public enum CustomerApprovalStep {
        TRANSLATION,
        RETURN
    }

    public enum CustomerApprovalTool {
        AMARA,
        CIELO24
    }

    public enum Language {
        @SerializedName("en")
        ENGLISH,
        @SerializedName("fr")
        FRENCH,
        @SerializedName("es")
        SPANISH,
        @SerializedName("de")
        GERMAN,
        @SerializedName("cmn")
        MANDARIN_CHINESE,
        @SerializedName("pt")
        PORTUGUESE,
        @SerializedName("jp")
        JAPANESE,
        @SerializedName("ar")
        ARABIC,
        @SerializedName("ko")
        KOREAN,
        @SerializedName("zh")
        TRADITIONAL_CHINESE,
        @SerializedName("hi")
        HINDI,
        @SerializedName("it")
        ITALIAN,
        @SerializedName("ru")
        RUSSIAN,
        @SerializedName("tr")
        TURKISH,
        @SerializedName("he")
        HEBREW;

        @Override
        public String toString() {
            return Utils.getSerializedName(this);
        }
    }
    
    public enum IWP {
        PREMIUM,
        INTERIM_PROFESSIONAL,
        PROFESSIONAL,
        SPEAKER_ID,
        FINAL,
        MECHANICAL,
        CUSTOMER_APPROVED_RETURN,
        CUSTOMER_APPROVED_TRANSLATION
    }

    public enum JobDifficulty {
        @SerializedName("Good")
        GOOD,
        @SerializedName("Bad")
        BAD,
        @SerializedName("Unknown")
        UNKNOWN;

        @Override
        public String toString() {
            return Utils.getSerializedName(this);
        }
    }
}

File: cielo24_PHP/src/cielo24/Enums.php
<?php

class TaskType
{
    const JOB_CREATED = "JOB_CREATED";
    const JOB_DELETED = "JOB_DELETED";
    const JOB_ADD_MEDIA = "JOB_ADD_MEDIA";
    const JOB_ADD_TRANSCRIPT = "JOB_ADD_TRANSCRIPT";
    const JOB_PERFORM_TRANSCRIPTION = "JOB_PERFORM_TRANSCRIPTION";
    const JOB_PERFORM_PREMIUM_SYNC = "JOB_PERFORM_PREMIUM_SYNC";
    const JOB_UPDATE_ELEMENTLIST = "JOB_UPDATE_ELEMENTLIST";
    const JOB_GET_TRANSCRIPT = "JOB_GET_TRANSCRIPT";
    const JOB_GET_CAPTION = "JOB_GET_CAPTION";
    const JOB_GET_ELEMENTLIST = "JOB_GET_ELEMENTLIST ";
}

class ErrorType
{
    const LOGIN_INVALID = "LOGIN_INVALID";
    const ACCOUNT_EXISTS = "ACCOUNT_EXISTS";
    const ACCOUNT_DOES_NOT_EXIST = "ACCOUNT_DOES_NOT_EXIST";
    const ACCOUNT_UNPRIVILEGED = "ACCOUNT_UNPRIVILEGED";
    const BAD_API_TOKEN = "BAD_API_TOKEN";
    const INVALID_QUERY = "INVALID_QUERY";
    const INVALID_OPTION = "INVALID_OPTION";
    const INVALID_URL = "INVALID_URL";
    const MISSING_PARAMETER = "MISSING_PARAMETER";
    const NOT_IMPLEMENTED = "NOT_IMPLEMENTED";
    const ITEM_NOT_FOUND = "ITEM_NOT_FOUND";
    const INVALID_RETURN_HANDLERS = "INVALID_RETURN_HANDLERS";
    const NOT_PARENT_ACCOUNT = "NOT_PARENT_ACCOUNT";
    const NO_CHILDREN_FOUND = "NO_CHILDREN_FOUND";
    const UNHANDLED_ERROR = "UNHANDLED_ERROR";
}

class JobStatus
{
    const Authorizing = "Authorizing";
    const Pending = "Pending";
    const In_Process = "In Process";
    const Complete = "Complete";
}

class TaskStatus
{
    const COMPLETE = "COMPLETE";
    const INPROGRESS = "INPROGRESS";
    const ABORTED = "ABORTED";
    const FAILED = "FAILED";
}

class Priority
{
    const ECONOMY = "ECONOMY";
    const STANDARD = "STANDARD";
    const PRIORITY = "PRIORITY";
    const CRITICAL = "CRITICAL";
    const HIGH = "HIGH";
}

class Fidelity
{
    const MECHANICAL = "MECHANICAL";
    const HIGH = "HIGH";
    const EXTERNAL = "EXTERNAL";
    const PREMIUM = "PREMIUM";
    const PROFESSIONAL = "PROFESSIONAL";
}

class CaptionFormat
{
    const SRT = "SRT";
    const SBV = "SBV";
    const DFXP = "DFXP";
    const QT = "QT";
    const TRANSCRIPT = "TRANSCRIPT";
    const TWX = "TWX";
    const TPM = "TPM";
    const WEB_VTT = "WEB_VTT";
    const ECHO_FORMAT = "ECHO";
}

class TokenType
{
    const WORD = "word";
    const PUNCTUATION = "punctuation";
    const SOUND = "sound";
}

class Tag
{
    const UNKNOWN = "UNKNOWN";
    const INAUDIBLE = "INAUDIBLE";
    const CROSSTALK = "CROSSTALK";
    const MUSIC = "MUSIC";
    const NOISE = "NOISE";
    const LAUGH = "LAUGH";
    const COUGH = "COUGH";
    const FOREIGN = "FOREIGN";
    const BLANK_AUDIO = "BLANK_AUDIO";
    const APPLAUSE = "APPLAUSE";
    const BLEEP = "BLEEP";
    const ENDS_SENTENCE = "ENDS_SENTENCE";
}

class SpeakerId
{
    const NO = "no";
    const NUMBER = "number";
    const NAME = "name";
}

class SpeakerGender
{
    const UNKNOWN = "UNKNOWN";
    const MALE = "MALE";
    const FEMALE = "FEMALE";
}

class TextCase
{
    const UPPER = "upper";
    const LOWER = "lower";
    const UNCHANGED = "";
}

class LineEnding
{
    const UNIX = "UNIX";
    const WINDOWS = "WINDOWS";
    const OSX = "OSX";
}

class CustomerApprovalSteps
{
    const TRANSLATION_STEP = "TRANSLATION";
    const RETURN_STEP = "RETURN";
}

class CustomerApprovalTools
{
    const AMARA = "AMARA";
    const CIELO24 = "CIELO24";
}

class Languages
{
    const en = "en";
    const fr = "fr";
    const es = "es";
    const de = "de";
    const cmn = "cmn";
    const pt = "pt";
    const jp = "jp";
    const ar = "ar";
    const ko = "ko";
    const zh = "zh";
    const hi = "hi";
    const it = "it";
    const ru = "ru";
    const tr = "tr";
    const he = "he";
}

class IWP
{
    const PREMIUM = "PREMIUM";
    const INTERIM_PROFESSIONAL = "INTERIM_PROFESSIONAL";
    const PROFESSIONAL = "PROFESSIONAL";
    const SPEAKER_ID = "SPEAKER_ID";
    const FINAL_ = "FINAL";
    const MECHANICAL = "MECHANICAL";
    const CUSTOMER_APPROVED_RETURN = "CUSTOMER_APPROVED_RETURN";
    const CUSTOMER_APPROVED_TRANSLATION = "CUSTOMER_APPROVED_TRANSLATION";
}